src/main/java/core/basesyntax/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Storage storage) {
+        storage.removeFruit(transaction.getFruit(), transaction.getQuantity());
+    }
```

Comment:
Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Currently, there is no check for a negative balance after the purchase operation, which could lead to incorrect storage state.